### DISCUSSION TITLE
Is there anyway to increment the build number during the upload_to_testflight lane?

### DISCUSSION THREAD

**Discussion by HenryGrange** [fastlane/fastlane#22050] [Q&A]
Hi,

I am creating an iOS pipeline in our gitlab repository. Between the build and upload job I have a wrapping job that wraps the .ipa for additional security (required). That would mean that if i use `increment_build_number(latest_testflight_build_number + 1)` in the build job, then the same build number could be assigned to two different jobs if two developers triggered the build job within ~30min of each other. To try and solve this, I tried updating the build number during the upload_to_testflight lane instead of during the build lane by setting the `build_number` parameter of `upload_to_testflight` job to `increment_build_number(latest_testflight_build_number + 1)` but it doesn't work. It still assigns the build_number it inherits from Xcode. 

Is there anyway to increment the build_number during the upload_to_testflight lane instead of during the gym or build lane? 

Thanks

**Comment by gabrielbragapro** [comment]
Isn't it supposed to be `latest_testflight_build_number` instead of `latest_testflight_number`?

**Reply by HenryGrange** [reply]
Yes, I typed this question in a hurry. Thanks for the correction I edited the post. But the point of this question still stands.
